docs/maven-tutorial.md

```diff
@@ -0,0 +1,218 @@
+# OpenShift + Maven Tutorial
+
+While developing your application with Java, you'll most likely be building
+many, many, many times.  In order to shorten the build times of your pods,
+dependencies can be cached in local Maven repositories.  This tutorial will step
```

Comment:
it actually doesn't die when the pod dies, it ends up baked into your application image and you can leverage it on future builds if you use incremental builds, which is something we should mention in this doc(just as a "btw if you care build performance you should also look at..")